### DISCUSSION TITLE
How to get Model activities between two timestamps ?

### DISCUSSION THREAD

**Discussion by pkd2989** [laravel/framework#43398] [Q&A]
Hi, I need some guidance.

I want to keep my Local app's database table synced with the remote Laravel server's table. The local app table is my Android Room database table.

So Basically, for a Model, I want to get the changes/events(addition, deletion, updation) that has taken place in-between the two timestamps.

Is there any builtin option for such thing or should I keep track record of Model Events. 

Thanks,

**Comment by henzeb** [comment]
Maybe https://spatie.be/docs/laravel-activitylog/v4/introduction is something for you, as it can log model events for you.

**Reply by pkd2989** [reply]
I can use normal Model events if I have to, and store in a separate table the ids of records that has been changed with the timestamps. I was wondering if something already exists that can give me the "changes" between two timestamps.

If not, there might be an opportunity to create a Model Trait "trackable", which can be applied to any model. Then there will be methods that can take two timestamp values can return the record ids and change type. This information can be used to remote application to update itself.

Thanks

**Reply by henzeb** [reply]
activity-log is tracking it. Not sure if they have methods for finding changes.
